clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -3021,6 +3021,12 @@ void prepareTypeConverter(mlir::LLVMTypeConverter &converter,
   converter.addConversion([&](mlir::cir::DoubleType type) -> mlir::Type {
     return mlir::FloatType::getF64(type.getContext());
   });
+  converter.addConversion([&](mlir::cir::FP16Type type) -> mlir::Type {
+    return mlir::FloatType::getF16(type.getContext());
```

Comment:
I'd like to see LLVM lowering tests for all these new types.

Reply:
The LLVM lowering for the types themselves is covered in [clang/test/CIR/Lowering/float.cir](https://github.com/llvm/clangir/pull/571/files#diff-64dcb563e6b26fa528a040c561b53ad425ab85891a369a425f04418134091082) .